src/reducers/url-state.js

```diff
@@ -395,20 +395,25 @@ const hiddenLocalTracksByPid: Reducer<Map<Pid, Set<TrackIndex>>> = (
       return hiddenLocalTracksByPid;
     }
     case 'SHOW_PROVIDED_TRACKS': {
-      const hiddenLocalTracksByPid = new Map();
-      // Go through the local tracks and make the provided ones visible.
-      for (const [pid, hiddenLocalTracks] of state.entries()) {
-        const newHiddenLocalTracks = new Set(hiddenLocalTracks);
-        // Remove all provided local tracks.
-        const localTracks = action.localTracksByPidToShow.get(pid);
-
-        if (!localTracks) {
+      const hiddenLocalTracksByPid = new Map(state);
+      // Go through the filtered local tracks to make them visible.
+      for (const [
+        pid,
+        localTracksToMakeVisible,
+      ] of action.localTracksByPidToShow.entries()) {
+        const hiddenLocalTracks = state.get(pid);
+        if (!hiddenLocalTracks) {
           throw new Error('Failed to find the local tracks');
         }
```

Comment:
I was wondering if we were sure that all pid are in the state, but it looks like so after studying the code. But are we sure this invariant will stay in the future, given that we don't enforce it really?
I think I'd be less worried if we didn't throw but instead were creating a new Set :-)

Reply:
I'm okay with either way really. Changed it.